### PR TITLE
[Feature] Add Variable for Storing XProtocol Req/Resp Raw Data

### PR DESCRIPTION
**Pull request by antJack**
related issue: https://github.com/mosn/mosn/issues/1965

### CODE REVIEW

File: pkg/protocol/xprotocol/bolt/decoder.go [hunk 1]
```diff
@@ -72,6 +74,7 @@ func decodeRequest(ctx context.Context, data api.IoBuffer, oneway bool) (cmd int
 	//4. copy data for io multiplexing
 	request.Data.Write(bytes[:frameLen])
 	request.rawData = request.Data.Bytes()
+	ctx = mosnCtx.WithValue(ctx, types.ContextKeyRequestRawData, request.rawData)
```
Comment (taoyuanyuan) [inline pkg/protocol/xprotocol/bolt/decoder.go:77]: Why not use variables?

Comment (antJack) [reply pkg/protocol/xprotocol/bolt/decoder.go]: A huge mistake

**Review by wangfakang — APPROVED** [review]
LGTM
BTW the values of other newly added variables can be obtained via streamBuffersByContext for request or response information.

**Review by wangfakang — APPROVED** [review]

**Review by doujiang24 — APPROVED** [review]
We'd better add tests for new features.

**Pull request merged**